docs/source/notes/cuda.rst

```diff
@@ -513,3 +513,189 @@ by GIL of Python interpreter.
 
 If you use :class:`~torch.nn.parallel.DistributedDataParallel`, you could use
 `torch.distributed.launch` utility to launch your program, see :ref:`distributed-launch`.
+
+.. _cuda-graph-semantics:
+
+CUDA Graphs
+-----------
+
+A graph is a frozen record of CUDA work (mostly kernels and their arguments).
+For general principles and details on the underlying CUDA API, see
+`Getting Started with CUDA Graphs`_ and the
+`Graphs section`_ of the CUDA C Programming Guide.
+
+The Pytorch integration creates graphs using `stream capture`_, which puts a CUDA stream
+in *capture mode*. CUDA work issued to a capturing stream doesn't actually run on
+the GPU. Instead, the work is recorded in a graph.
+
+After capture, the graph can be *launched* to run the GPU work as many times as needed.
+Each replay runs the same frozen ops on the same memory addresses.
+By filling input memory with new data (e.g., from a new batch) before each replay,
+you can rerun the same work on new data.
+
+Replay sacrifices the dynamic flexibility of typical eager execution in exchange for
+**greatly reduced CPU overhead**.
```

Comment:
suggested new sentence to end this paragraph: Graph replay skips all layers of kernel dispatch, including Python, C++, and CUDA driver overheads.